### ISSUE TITLE
Notes on Testing Dates

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#4558]
TL;DR, our tests crash if you create a PR at 23:55 because our tests have logic like "# of patients registered in the last day" which may be triggered at 00:05 the follow day and produce an unexpected result.

I've noticed that with Dependabot sending PRs are midnight, our tests often crash.  This is because we have [many tests that depend on relative dates](https://github.com/IMA-WorldHealth/bhima/blob/master/test/end-to-end/patient/registry.search.js#L83).  It occurs to me that this doesn't make sense.  We can't predict when the tests will run, so relative dates such as "in the last year" or "in the last day" don't make sense _unless we specify the time of the server_ (e.g. tell the server that is 12:33:15 on 03/05/2016).

Pegging the server time doesn't make sense for local testing (in my opinion) unless something like sinon.js's fake timers provide that functionality easily.  Either way, it is clear we should separate those timing tests from the rest of the test suite so that we can have greater predictability in our testing.
